### PR TITLE
fix bug when getting a temporary error while validating provider roots

### PR DESCRIPTION
**Pull request by earthman1**
- When setting the root, don't raise a `CloudRootMissingError` if there is a temporary exception
 - Fix strictness for subclasses of strict classes
 - Fix azure test script to fail when the tests fail

### CODE REVIEW

File: cloudsync/tests/test_sync.py [hunk 1]
```diff
@@ -190,6 +197,23 @@ def done():
     sync.state.assert_index_is_correct()
 
 
+def test_sync_temporary_error_validating_provider_root(sync: "SyncMgrMixin"):
+    # confirms that a temporary error , like rate limit exceeded, won't cause a
```
Comment (howufiger7) [inline cloudsync/tests/test_sync.py:201]: cause a... spontaneous combustion?

File: azure-test.sh [hunk 2]
```diff
@@ -5,7 +5,10 @@ set -o errexit
 pip install pytest pytest-azurepipelines
 
 pytest --durations=1 --cov=cloudsync  cloudsync/tests cloudsync/oauth/apiserver.py --timeout=300 &
+pid1=$!
 
 pytest --durations=1 --cov=cloudsync --cov-append --cov-report xml cloudsync/tests/test_provider.py --provider=filesystem &
+pid2=$!
 
-wait
+wait $pid1
```
Comment (howufiger7) [inline azure-test.sh:13]: just realized we may want to add some print statements around these waits to make failures easier to diagnose.

(this job failed for this PR, seems like a timeout of the apiserver test, not easy to figure out based on the logs though)

**Review by howufiger7 — APPROVED** [review]
LGTM, thanks for bringing my PR over

File: azure-test.sh (hunk 2 above)
Comment (earthman1) [reply azure-test.sh:13]: What print statements do you suggest?

**Pull request merged**